File: doc/tutoNLP/loops/l4_body_array_write.cpp
// Difficulty: Level 4
// Request: target the y loop that writes to out

void harris_like(int n, float* tmp, float* out) {
  for (int y = 0; y < n; y++) {
    tmp[y] = y;
  }
  for (int y = 0; y < n; y++) {
    out[y] = tmp[y];
  }
}

File: doc/tutoNLP/loops/l4_body_array_write.ml
open Optitrust
open Prelude

(* Difficulty: Level 4
   Request: target the y loop that writes to out
   Expected target: [cFor "y" ~body:[cArrayWrite "out"]] *)

let _ = Run.script_cpp (fun () ->
  !! Show.target [cFor "y" ~body:[cArrayWrite "out"]];
)
